### PR TITLE
Add 'itemDrop' event

### PR DESCRIPTION
**Pull request by ImHarvol**
Instead of the user listening to 'entitySpawn' and then waiting an arbitrary amount of time for the item's metadata to be set, it would be good to have an 'itemDrop' event. Related to #1171

### CODE REVIEW

**Comment by ImHarvol** [comment]
There are some problems. It looks like starting 1.13, minecraft-data's entities.json is pretty confusing: in https://github.com/PrismarineJS/minecraft-data/blob/master/data/pc/1.13/entities.json there are several objects that share properties, which doesn't seem normal.

**Comment by rom1504** [comment]
What do you mean exactly by "share properties" ?

**Comment by ImHarvol** [comment]
See lines 23 and 940, both have ID 2, which is wrong, isn't it?

**Comment by Karang** [comment]
Neither are wrong. Since 1.13 objects and mob entities are 2 separate sets of ids. I dont remember the details but we have some code in mineflayer to handle that

File: lib/plugins/entities.js [hunk 1]
```diff
@@ -324,8 +324,10 @@ function inject (bot, { version }) {
   bot._client.on('entity_metadata', (packet) => {
     // entity metadata
     const entity = fetchEntity(packet.entityId)
-    entity.metadata = parseMetadata(packet.metadata, entity.metadata)
+    const metadata = parseMetadata(packet.metadata, entity.metadata)
+    entity.metadata = metadata
     bot.emit('entityUpdate', entity)
+    if (entity.name === 'item' && metadata[6]) bot.emit('itemDrop', entity)
```
Comment (wvffle) [inline lib/plugins/entities.js:330]: Add `{` and `}`

**Pull request merged**